### DISCUSSION TITLE
How to configure some packages to be updated more often than "general" schedule?

### DISCUSSION THREAD

**Discussion by stasostrovskyi** [renovatebot/renovate#10643] [Request Help] [answered]
We are running a self-hosted renovate and trying to set it up for a bunch of languages/managers. Our idea is to update repositories weekly (by following `["schedule": "weekly"]`).  However, we would also like to bump the version of some "critical" own libs as soon as they are updated, for example following `["schedule": "daily"]`. We are using our own shareable preset and all the repositories refer to it. This is a snippet from the config I've set up, but it doesn't seem like renovate updates anything `daily`: 

```
{
    "extends": [
        "config:base",
        "schedule:weekly",
    ],
    "packageRules": [
        {
            "matchManagers": [
                "maven"
            ],
            "registryUrls": [
                "some-url"
            ],
            "groupName": "maven-deps",
            "commitMessageTopic": "maven dependencies"
        },
        {
            "matchPackageNames": ["critical-lib1",
                                                       "critical-lib2"],
            "extends": ["schedule:daily"]
        },
       //...other rules
    ]
}
```

How does schedule "merging" work? What happens if I have one global schedule, one inside the rule, and for example one for a specific package manager? What is the best way to renovate some packages more often then regular updates?

**Comment by rarkins** [comment, marked as answer]
You'll need to be more specific than "doesn't seem like"

**Reply by stasostrovskyi** [reply]
Sorry for that. 
What I see in the logs, update of a "critical" lib is grouped with other ones and that's how renovate is trying to schedule it

```
[2021-06-29T09:05:33.497Z] DEBUG: Checking schedule(before 3am on Monday, null) 
[2021-06-29T09:05:33.497Z] DEBUG: Checking 1 schedule(s)
[2021-06-29T09:05:33.497Z] DEBUG: Checking schedule "before 3am on Monday" 
[2021-06-29T09:05:33.497Z]        "parsedSchedule": {"schedules": [{"t_b": [10800], "d": [2]}], "exceptions": [], "error": -1}
[2021-06-29T09:05:33.497Z] DEBUG: Package not scheduled 
[2021-06-29T09:05:33.497Z] DEBUG: Skipping branch creation as not within schedule
```

**Reply by rarkins** [reply]
It sounds like you have a group (MR) of updates, some with schedule weekly and some with schedule monthly? What were you hoping to happen in such a case?

**Reply by viceice** [reply]
Every group should have it's own schedule, otherwise i think renovate will use the schedule from first update in group.

**Reply by stasostrovskyi** [reply]
No, I don't have any groups with monthly schedule. I only have weekly schedule on top of config file outside any groups. 

What I'm trying to achieve is - "I want all my updates, for any package managers, happen on weekly schedule, also I want my maven packages all be grouped together, but for maven libs A,B,C I want to receive updates daily". Is this even possible or I want too much? :)

**Reply by viceice** [reply]
Then you need to use a separate group for your maven libs A,B,C.

**Reply by rarkins** [reply]
Either exclude them from the first rule group or set a different groupName (including possibly `null`) in the second rule. If you have mixed schedules in a single group then it explains the behaviour you're seeing.

**Reply by stasostrovskyi** [reply]
Got it. Thank you very much for your help! And renovate is truly a great tool, thank you for all your hard work!
